modules/nw-ipfailover-configuring-check-notify-scripts.adoc

```diff
@@ -0,0 +1,103 @@
+////
+
+When a check script is not provided, a simple default script is run that tests the TCP connection. This default test is suppressed when the monitor port is `0`.
+
+For each virtual IP (VIP), Keepalived keeps the state of the node. The VIP on the node may be in `master`, `backup`, or `fault` state. All VIPs on the node that are not in the `fault` state participate in the negotiation to decide which is `master` for the VIP. All of the losers enter the `backup` state. When the check script on the `master` fails, the VIP enters the `fault` state and triggers a renegotiation. When the `backup` fails, the VIP enters the `fault` state. When the check script passes again on a VIP in the `fault` state, it exits `fault` and negotiates for `master`. The resulting state is either `master` or `backup`.
+
+As cluster administrator, you can provide an optional notify script, which is called whenever the state changes. Keepalived passes the following three parameters to the script:
+
+* `$1` - `group` or `instance`
+* `$2` - Name of the `group` or `instance`
+* `$3` - The new state: `master`, `backup`, or `fault`
+
+These scripts run in the IP failover pod and use the pod file system, not the host file system. The options require the full path to the script. You must make the script available in the pod to extract the results from running the notify script. The recommended approach for providing the scripts is to use a config map.
+
+The full path names of the check and notify scripts are added to the Keepalived configuration file, `_/etc/keepalived/keepalived.conf`, which is loaded every time Keepalived starts. The scripts can be added to the pod with a config map as follows.
+
+.Prerequisites
+
+* You installed the OpenShift CLI (`oc`).
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create the desired script and create a config map to hold it. The script has no input arguments and must return `0` for `OK` and `1` for `fail`.
++
+The check script, `_mycheckscript.sh_`:
++
+[source,bash]
+----
+#!/bin/bash
+    # Whatever tests are needed
+    # E.g., send request and verify response
+exit 0
+----
+
+. Create the config map:
++
+[source,terminal]
+----
+$ oc create configmap mycustomcheck --from-file=mycheckscript.sh
+----
+
+. There are two approaches to adding the script to the pod: use `oc` commands or edit the deployment configuration. In both cases, the `defaultMode` for the mounted config map files must allow execution. A value of `0755`, `493` decimal, is typical.
+
+.. Using `oc` commands:
++
+[source,terminal]
+----
+$ oc env dc/ipf-ha-router \
+    OPENSHIFT_HA_CHECK_SCRIPT=/etc/keepalive/mycheckscript.sh
+$ oc set volume dc/ipf-ha-router --add --overwrite \
+    --name=config-volume \
+    --mount-path=/etc/keepalive \
+    --source='{"configMap": { "name": "mycustomcheck", "defaultMode": 493}}'
```

Comment:
The example yaml defines a deployment; should these `oc` commands likewise use a deployment rather than a deploymentconfig?

Reply:
I'm not certain. @rfredette would you mind giving this procedure a review, please?